circuits/cpp/barretenberg/ts/cjs-entry/index.cjs

```diff
@@ -0,0 +1,10 @@
+let loadedModule;
+
+async function loadModule() {
+  if (!loadedModule) {
+    loadedModule = await import('../dest/node/index.js');
+  }
+  return loadedModule;
+}
+
+module.exports = { loadModule };
```

Comment:
This approach will cause inconsistent API surface.
Once would expect to require from package in the same way between cjs and esm.

Reply:
That's right.
It maybe possible to "degrade" the esm experience to the requirement of having to call loadModule, but we shouldn't do this to support an old system for the sake of api consistency here.
If there is a *quick and clean* (e.g. 1 day of effort) solution then please do as separate PR. Otherwise the effort and ongoing build complexity our engineers are exposed to isn't justified.
Just a reminder, browsers and node are ESM compatible. The requirement for CJS support is to support a handful of frameworks that have not yet upgraded to ESM. This need for 'loadModule' is a temporary stop-gap for users that are still using these lagging frameworks.
I think this demonstrates our values of "good enough" and not investing further time in something for the sake of perfection.
The README makes the esm vs cjs import approach clear at least.